### ISSUE TITLE
propsal: Implement FigletPrinter as an alternative to BigTextPrinter

### ISSUE THREAD

**Issue by tw1nk** [pterm/pterm#219]
The FigletPrinter could use alternative figlet font files. 

There is a go module to read figlet font files
https://github.com/lukesampson/figlet/figletlib
which can parse the figlet font file format. (https://github.com/lukesampson/figlet/blob/master/figfont.txt) 

You can read more about figlet fonts and see examples at: 
http://www.figlet.org/

I'm willing to add this feature

**Comment by MarvinJWendt** [comment]
Hi @tw1nk, thanks for the suggestion! I didn't knew about figlet font files, but it would totally make sense to implement them. Currently, you can change the "Font" of the `BigTextPrinter` via `WithBigCharacters(m map[string]string)`. So in theory, we would just need a function to convert figlet font files, into a go map. There are three possiblities now:

1. Add the converter function to the `putils` package 
    - `pterm.DefaultBigText.WithBigCharacters(putils.BigCharactersFromFigletFont("font")).Render()`
1. Add the converter function to the PTerm directly
      - `pterm.DefaultBigText.WithBigCharacters(pterm.BigCharactersFromFigletFont("font")).Render()`
1. Add a new `WithXxx` function to the `BigTextPrinter`
     - `pterm.DefaultBigText.WithBigCharactersFromFigletFont("font").Render()`

The problem with 2. is that it would clutter PTerm. We try to get rid of global functions, that only help in a specifc printer, because that exposes too many functions over the long run. We'll probably deprecate "LettersFromString" and move that function into `putils` too, but I am not sure atm.  
The third point is better in my opinion, because it wouldn't expose a new global function. But it would also add an option, which is a wrapper for another option, and I don't really know what to think about that. Also `WithBigCharacters` and `WithBigCharactersFromFigletFont` would not be combinable, and at the moment we don't have options, which aren't combinable, in any printer.   
The first point is my personal favourite, as we can add as many functions to `putils` as we want, and we don't really have to care how many exports there are. `putils` will contain many similar functions in the future, so I guess that's the best fit to put it there. `putils` will contain functions like `TableFromCSV`, `TableFromJSON`, `TableFromStructSlice`, etc. soon.

So I would go with 1.  
If I missed something, let me know!  
Feel free to open a PR (or draft PR) and I'll asign you to this issue :)

**Comment by tw1nk** [comment]
Yeah, that was my thoughts as well. I'll go for the putils route. 

That will also make it easier to move utility functions to separate modules in the future to limit the number of external dependencies.
